pootle/core/utils/version.py

```diff
@@ -0,0 +1,232 @@
+    '1.2'
+    """
+    version = get_complete_version(version)
+    return _get_version_string(version[:2])
+
+
+def get_complete_version(version=None):
+    """Returns a tuple of the Pootle version. Or the supplied ``version``
+
+    >>> get_complete_version((1, 2, 3, 'alpha', 0))
+    (1, 2, 3, 'alpha', 0)
+    """
+    if version is None:
+        from pootle import VERSION as version
+
+    return version
+
+
+def get_docs_version(version=None, positions=2):
+    """Return the version used in documentation.
+
+    >>> get_docs_version((1, 2, 1, 'alpha', 0))
+    'dev'
+    >>> get_docs_version((1, 2, 1, 'rc', 2))
+    '1.2'
+    """
+    version = get_complete_version(version)
+    candidate = _get_candidate(version)
+    candidate_pos = _get_candidate_pos(version)
+    if positions > candidate_pos:
+        positions = candidate_pos
+    if _is_development_candidate(version):
+        return 'dev'
+    return _get_version_string(version[:positions])
+
+
+def _shell_command(command):
+    """Return the first result of a shell ``command``"""
+    repo_dir = os.path.dirname(os.path.abspath(__file__))
+
+    command_subprocess = subprocess.Popen(
+        command,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+        cwd=repo_dir,
+        universal_newlines=True
+    )
+
+    return command_subprocess.communicate()[0]
+
+
+@lru_cache()
+def get_git_changeset():
+    """Returns a numeric identifier of the latest git changeset.
+
+    The result is the UTC timestamp of the changeset in YYYYMMDDHHMMSS format.
+    This value isn't guaranteed to be unique, but collisions are very unlikely,
+    so it's sufficient for generating the development version numbers.
+
+    >>> get_git_changeset()
+    '20150530132219'
+    """
+    timestamp = _shell_command(
+        ['git', 'log', '--pretty=format:%ct', '--quiet', '-1', 'HEAD']
+    )
+    try:
+        timestamp = datetime.datetime.utcfromtimestamp(int(timestamp))
+    except ValueError:
+        return None
+    return timestamp.strftime('%Y%m%d%H%M%S')
+
+
+@lru_cache()
+def get_git_branch():
+    """Returns the current git branch.
+
+    >>> get_git_branch()
+    'feature/proper_version'
+    """
+    branch = _shell_command(['git', 'symbolic-ref', '-q', 'HEAD']).strip()
+    if not branch:
+        return None
+    return "/".join(branch.split("/")[2:])
+
+
+@lru_cache()
+def get_git_hash():
+    """Returns the current git commit hash.
+
+    >>> get_git_hash()
+    'ad768e8'
+    """
+    return _shell_command(
+        ['git', 'rev-parse', '--verify', '--short', 'HEAD']
+    ).strip()
+
+if __name__ == "__main__":
+    from sys import argv
+    if len(argv) == 2:
+        if argv[1] == "main":
+            print get_main_version()
+        elif argv[1] == "major_minor":
+            print get_major_minor_version()
+        elif argv[1] == "docs":
+            print get_docs_version()
+    else:
+        print get_version()
```

Comment:
For future reference, don't forget to add parentheses to print statements, even if you don't import print_function. It's free and it's better. :)